src/ParametricFunctionControlPanel.java

```diff
@@ -0,0 +1,215 @@
+        JTextField rangeParameter = new JTextField("< = t < =");
+        rangeParameter.setEditable(false);
+        rangeParameter.setHorizontalAlignment(JTextField.CENTER);
+        rangeParameter.setFont(new Font("Arial", Font.PLAIN, 20));
+        gbc.gridx = 2;
+        gbc.gridy = 3;
+        gbc.gridwidth = 1;  //Could be deleted
+        gbc.ipady = 0;      //Could be deleted
+        gbc.weightx = 1;    //Could be deleted
+        ParametricInputPosition.setConstraints(rangeParameter, gbc);
+        add(rangeParameter);
+
+        //Creates the second Box to declare the Range
+        rangeEnd.setPreferredSize(new Dimension(80, 20));
+        rangeEnd.setHorizontalAlignment(JTextField.CENTER);
+        rangeEnd.setFont(new Font("Arial", Font.PLAIN, 16));
+        gbc.gridx = 3;
+        gbc.gridy = 3;
+        gbc.gridwidth = 1;  //Could be deleted
+        gbc.ipady = 20;
+        gbc.weightx = 1;    //Could be deleted
+        ParametricInputPosition.setConstraints(rangeEnd, gbc);
+        add(rangeEnd);
+        AbstractDocument RangeEndDoc = (AbstractDocument) rangeEnd.getDocument(); //Sets a DocumentFilter to the TextField to only allow numeric input
+        RangeEndDoc.setDocumentFilter(new RangeNumericFilter());
+
+    }
+
+    private void createDeleteButton() {
+        JButton delete = new JButton("Delete");
+        delete.setPreferredSize(new Dimension(80, 20));
+        delete.setFocusable(false);
+        delete.addActionListener(new DeleteActionListener());
+        gbc.gridx = 4;
+        gbc.gridy = 3;
+        gbc.gridwidth = 1;  //Could be deleted
+        gbc.ipady = 0;
+        gbc.weightx = 1;    //Could be deleted
+        ParametricInputPosition.setConstraints(delete, gbc);
+        add(delete);
+    }
+
+
+    private void createCalculateButton() {
+        JButton calculate = new JButton("Calculate");
+        calculate.setPreferredSize(new Dimension(80, 20));
+        calculate.setFocusable(false);
+        calculate.addActionListener(new CalculateActionListener());
+        gbc.gridx = 5;
+        gbc.gridy = 3;
+        gbc.gridwidth = 1;  //Could be deleted
+        gbc.ipady = 0;      //Could be deleted
+        gbc.weightx = 1;    //Could be deleted
+        ParametricInputPosition.setConstraints(calculate, gbc);
+        add(calculate);
+    }
+
+    private static class RangeNumericFilter extends DocumentFilter {
+
+        @Override
+        public void insertString(FilterBypass fb, int offset, String range, AttributeSet attr) throws BadLocationException {
+
+            if (isNumeric(range))
+                super.insertString(fb, offset, range, attr);
+        }
+
+        @Override
+        public void replace(FilterBypass fb, int offset, int length, String range, AttributeSet attrs) throws BadLocationException {
+
+            if (isNumeric(range))
+                super.replace(fb, offset, length, range, attrs);
+        }
+
+        public boolean isNumeric(String range) {
```

Comment:
I removed the variable and moved the return statement.
But I do not understand the need to check for an empty String. This function only checks user input and the user can not input "empty". The only thing the user can do is delete the complete String written in it, which is handled by the replace function.

Reply:
I didnt look completely at the code. What happens when a user inserts a empty string? Because if he does it is not null it just has the length of 0 so "". Does it get filterd beforehand?

Reply:
Done